src/cairo_run.rs

```diff
@@ -77,109 +72,103 @@ pub fn cairo_run(
     }
     cairo_runner.relocate(&mut vm)?;
 
-    if cairo_run_config.print_output {
-        write_output(&mut cairo_runner, &mut vm)?;
+    if let Some(output_writer) = output_writer {
+        cairo_runner.write_output(&mut vm, output_writer)?;
```

Comment:
Maybe we should just remove this and expect `cairo_run`'s caller to call `write_output` if they want. That would simplify the call in main as it would just be checking if `print_output` is set in the command line flags after executing `cairo_run`.

Reply:
Totally. I would be more than happy with this.
It would also prevent the prefix "Program Output" to be written to the bufwriter, even if the execution fails. Which was really ugly, but impossible to avoid if we kept passing the writer to the runner

Reply:
Except that they must be able to access the `vm` in order to read the builtin. And `vm` is no longer returned by `cairo_run`.

Btw, `write_output` does not use `self` at any point. It has absolutely no reason to be a `CairoRunner` method. It's actually a `VirtualMachine` method if any coz it needs a `&mut vm` parameter.

Reply:
Should I put the `vm` back in the return type of `cairo_run`? 

Another option is for `cairo_run` to return a tuple `(CairoRunner, Option<String>)` with the second one being the output if the flag has been passed to true and the builtin segment is present.

I prefer solution 1, but tell me what you want @Oppen

Reply:
I think the better option is number 1. I agree as well that if it doesn't use the runner it makes sense to make it a method for the VM.
I'd like the opinion of @fmoletta about this. I'll ping her.

Reply:
Sounds good, we can return a tuple `(CairoRunner, VirtualMachine)`, we had to remove the vm from the runner during the python integration and forgot to re-expose the vm after the run

Reply:
Excellent then. You can go ahead and do that @tdelabro and then we'll be finally done :)